src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -1036,7 +1037,10 @@ protected internal virtual unsafe void Reset(bool forceReleaseLongLivedAllocator
 
                 _pooledArrays = null;
             }
-        }
+            //TODO:Don't PR this, just testing.
```

Comment:
You forgot these?

Reply:
This PR is meant not for merging, it is more of a note to self what i have modified while playing with the code.
I'm not sure why i can't reply to the comment above, i'll reply here regarding Gen2.
What is going on is that memory grows pretty much linearly with the parallel level, on my machine each thread is able to peak 100MB per context. there are about 32 session threads so the 4GB of allocated memory is somewhat expected.
You're correct regarding the fact that we utilize the unmanaged memory once the session is disposed of the request executor with its context pool is returned to the document store, there is only one request executor in this case but the pool it owns has about 37 contexts in it each with 100MB.
So the blittable unmanaged memory isn't leaking!
The thing is we are generating _alot_ of reader objects and the stress on the system might be the cause of them to reach Gen2 but what i can't explain is why they are not all collected when i call GC.Collect, there are the ones held by the context in caches it is expected they won't be released but there seems to be a whole bunch of them without GC roots and they are the results of the save changes and indeed looking at the code they are not been refrenced, not sure why they are not collected every 32 sessions i invoked GC.Collect, i would expect none to survive...
I tested this on a new project, generated a bunch of unsafe object pointing to blobs of unmanaged memory and calling GC collect, one of them seems to always survive but the rest are been cleaned by the GC... I guess it is somehow been pointed by a temp variable on the stack generated by the JIT or something...

Reply:
One important thing I missed in my scroll above is the fact that we don't have low memory notification on the client so the ContextPool is never shrinking even after the "peak load" I think that introducing low memory machanizem on the client side even if as an optional thing could help us behave nicer on client machines.